packages/flutter/test/widgets/dispose_ancestor_lookup_test.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart' hide TypeMatcher;
+import 'package:flutter/widgets.dart';
+
+typedef void TestCallback(BuildContext context);
+
+class TestWidget extends StatefulWidget {
+  TestWidget(this.callback);
+
+  final TestCallback callback;
+
+  @override
+  TestWidgetState createState() => new TestWidgetState();
+}
+
+class TestWidgetState extends State<TestWidget> {
+  @override
+  void dispose() {
+    widget.callback(context);
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) => const Text('test');
+}
+
+void main() {
```

Comment:
That case is handled implicitly by most tests however I added an explicit test just to be complete.